_posts/minecraft/java/2024-07-16-server-properties.md

```diff
@@ -0,0 +1,339 @@
+
+---
+  
+### force-gamemode
+
+Possible Values: Boolean
+Default Value: false
+
+This setting forces all players to be switched to a specific game mode upon joining the server.
+
+---
+
+### gamemode
+
+Possible Values: survival, creative, adventure, spectator
+Default Value: survival
+
+This setting defines the default game mode for players joining the server, such as survival, creative, or adventure.
+
+- If set to `survival`, Players must gather resources, build structures, and survive against hostile mobs.
+- If set to `creative`, Players have unlimited access to the resources, They can fly, hostile mobs do not attack them.
+- If set to `adventure`, Players can interact with objects like levers and buttons, but they are unable to break the blocks.
+- If set to `spectator`, Players can fly, phase through blocks, see through blocks. They cannot interact with the world and are invisible to other players.
+
+---
+
+### generate-structures
+
+Possible Values: Boolean
+Default Value: true
+
+This setting controls whether structures like villages and temples generate in the world, which can impact gameplay and exploration.
+
+---
+
+### hardcore
+
+Possible Values: Boolean
+Default Value: false
+
+Setting this to `true` will change the server difficulty to hard and sets gamemode to spectator upon death if they choose to respawn, adding an extra layer of challenge.
+
+---
+
+### level-seed
+
+Possible Values: String
+Default Value: *blank*
+
+This setting allows specifying a seed for world generation, enabling the creation of specific world layouts or replicating previous worlds.
+
+---
+
+### log-ips
+
+Possible Values: Boolean
+Default Value: true
+
+This setting determines if player IP addresses are logged upon connection, which can be useful for monitoring and security purposes. Set this to `false` if you prefer the server to be privacy-friendly.
+
+---
+
+### max-players
+
+Possible Values: Integer
+Default Value: 20
+
+This setting specifies the maximum number of players allowed on the server simultaneously, impacting server load and community size.
+
+---
+
+### max-tick-time
+
+Possible Values: Integer (in milliseconds) or -1
+Default Value: 60000
+
+This setting defines the maximum time in milliseconds a single tick may take before the server watchdog stops the server, helping to prevent severe performance issues.
+
+- If a single server tick took 60 seconds, the watchdog will consider it as a crash and forcibly shut downs the server.
+- Set to `-1` to disable the watchdog.
+
+---
+
+### max-world-size
+
+Possible Values: Integer
+Default Value: 29999984
+
+The maximum allowed size of the world radius, in blocks. This only affects the chunks that are generated when the world is initially created, and not the world border.
+
+---
+
+### motd
+
+Possible Values: String
+Default Value: A Minecraft Server
+
+The message of the day (MOTD) is displayed in the server list and provides a brief description or welcome message for players.
```

Comment:
You meticulously crafted a guide for this!
Add the {.info} thing telling users to refer to your guide.